testing/server.go

```diff
@@ -126,3 +136,22 @@ func (s *Server) UploadCharm(c *gc.C, ch charm.Charm, id *charm.Reference, promu
 	}
 	return result.Id
 }
+
+// ChangeReadPerm changes read permissions of the given entity.
+func (s *Server) ChangeReadPerm(c *gc.C, id *charm.Reference, perms ...string) {
```

Comment:
I'd prefer to add this method (or something like it) to the cs client itself - it's actually useful functionality, and not just for testing.